src/services/UserInfoRequestProcessor.ts

```diff
@@ -0,0 +1,189 @@
+				accessTokenExpiryDate: session.accessTokenExpiryDate,
+				attemptCount: session.attemptCount,
+				authorizationCodeExpiryDate: session.authorizationCodeExpiryDate,
+				createdDate: session.createdDate,
+				expiryDate: session.expiryDate,
+				redirectUri: session.redirectUri,
+			},
+		});
+  
+		this.metrics.addMetric("found session data", MetricUnits.Count, 1);
+
+		const personInfo: PersonIdentityItem | undefined = await this.BavService.getPersonIdentityBySessionId(sessionId, this.personIdentityTableName);
+		if (!personInfo) {
+			this.logger.error("No person found with this session ID", {
+				messageCode: MessageCodes.PERSON_NOT_FOUND,
+			});
+			return new Response(HttpCodesEnum.UNAUTHORIZED, "Unauthorized");
+		}
+
+		this.logger.info("Found person Identity data");
+  
+		this.metrics.addMetric("found person identity data", MetricUnits.Count, 1);
+
+		// Validate the AuthSessionState to be "BAV_ACCESS_TOKEN_ISSUED"
+		if (session.authSessionState !== AuthSessionState.BAV_ACCESS_TOKEN_ISSUED) {
+			this.logger.error("Session is in wrong Auth state", {
+				// note: we only log specific non-PII attributes from the session object:
+				expectedSessionState: AuthSessionState.BAV_ACCESS_TOKEN_ISSUED,
+				session: {
+					authSessionState: session.authSessionState,
+				},
+				messageCode: MessageCodes.STATE_MISMATCH,
+			});
+			return new Response(HttpCodesEnum.UNAUTHORIZED, "Unauthorized");
+		}
+
+		// Person info required for VC
+		const names = personInfo.name[0].nameParts;
+
+		// Validate the User Info data presence required to generate the VC
+		if (names && names.length > 0 && session.sortCode && session.accountNumber) {
+
+			//Generate VC and create a signedVC as response back to IPV Core.
+			const signedJWT = await this.verifiableCredentialService.generateSignedVerifiableCredentialJwt(session, names, {
+				sortCode: session.sortCode,
+				accountNumber: session.accountNumber,
+			}, absoluteTimeNow);
+
+			// Add metric and send TXMA event to the sqsqueue
+			this.metrics.addMetric("Generated signed verifiable credential jwt", MetricUnits.Count, 1);
+
+			const txmaCoreFields = buildCoreEventFields(session, this.issuer, session.clientIpAddress, absoluteTimeNow);
+			
+			await this.BavService.sendToTXMA(
+				this.txmaQueueUrl, {
+					event_name: TxmaEventNames.BAV_CRI_VC_ISSUED,
+					...txmaCoreFields,
+					restricted: {
+						name: [{
+							nameParts: names,
+						},
+						],
+					},
+				});
+
+			await this.BavService.sendToTXMA(
+				this.txmaQueueUrl, {
+					event_name: TxmaEventNames.BAV_CRI_END,
+					...txmaCoreFields,
+				},
+			);
+			
+
+			// return success response
+			return new Response(HttpCodesEnum.OK, JSON.stringify({
+				sub: session.subject,
+				"https://vocab.account.gov.uk/v1/credentialJWT": [signedJWT],
+			}));
+		} else {
+			this.logger.error("Missing required fields to generate BAV VC", {
+				messageCode: MessageCodes.MISSING_PERSONAL_DETAILS,
```

Comment:
Do you think we can add the fields that are missing? Maybe something like 
```
{
  names: !!names,
  sortCode: !!session.sortCode,
  accountNumber: !!session.accountNumber
}
```
Or even using that obfuscate function you have in F2F?